docs/proposals/block-storage.md

```diff
@@ -25,90 +25,89 @@ lower bound).
 
 ## The requirements
 
-An obvious solution is **storing all blocks in RocksDB** (a key–value
-database that we are using). Indeed, a long time ago we had all blocks
-stored in RocksDB – but we switched from that to storing each block in a
-different file. Storing blocks in RocksDB has several related disadvantages:
+Here's a list of requirements for a good solution:
 
-  * If you want to back the database up you would have to copy the blocks as
-    well, while in reality the blocks can always be redownloaded and we
-    don't care much about losing them.
+  * R1. Store blocks separately from the main database.
 
-  * A hard drive failure has a much bigger chance to corrupt the database,
-    if the database contains lots of data (blocks).
+  * R2. Fast lookup by header hash.
+  
+  * R3. Fast sequential iteration
 
-So, here's a list of requirements for a good solution:
+  * R4. A solution should be stable to corruptions
 
-  * We'd like to store blocks separately from the main database.
+  * R5. Apply compression to the blocks, and since each block isn't
+        very well compressible on its own, the solution should use either
+        dictionary compression or compress all blocks as a whole.
```

Comment:
I don't think this is a requirement. It's a plausible design choice. The requirement is "reasonable" use of disk space.